test/oqs_test_libctx.c

```diff
@@ -0,0 +1,563 @@
+        goto out;
+    }
+
+    if ((idx = oqs_get_idx(sigalg_name)) >= OSSL_NELEM(info_classical)) {
+        goto out;
+    }
+
+    if (!((pubkey1_len == pubkey2_len) && (privkey1_len == privkey2_len))) {
+        goto out;
+    }
+
+    ret = memcmp(pubkey1, pubkey2, info_classical[idx].pubkey_len) == 0 &&
+          memcmp(privkey1, privkey2, info_classical[idx].privkey_len) == 0;
+
+    ret_rev = memcmp(pubkey1 + pubkey1_len - info_classical[idx].pubkey_len,
+                     pubkey2 + pubkey2_len - info_classical[idx].pubkey_len,
+                     info_classical[idx].pubkey_len) == 0 &&
+              memcmp(privkey1 + privkey1_len - info_classical[idx].privkey_len,
+                     privkey2 + privkey2_len - info_classical[idx].privkey_len,
+                     info_classical[idx].privkey_len) == 0;
+
+out:
+    free(pubkey1);
+    free(pubkey2);
+    free(privkey1);
+    free(privkey2);
+    return ret || ret_rev;
+}
+
+/** \brief Compares the classical KEM elemets of two Encaps/Decaps executions
+ *
+ * \param kemalg_name algorithm name.
+ * \param sec1 A shared secret.
+ * \param sec2 A shared secret.
+ * \param seclen Shared secret length.
+ * \param ct1 An encapsulation.
+ * \param ct1 An encapsulation.
+ * \param ctlen Encapsulation lentgh.
+ *
+ * \return 1 if the compare operation is successful, else 0. */
+static int oqs_cmp_kem_elems(const char *kemalg_name, const unsigned char *sec1,
+                             const unsigned char *sec2, size_t seclen,
+                             const unsigned char *ct1, const unsigned char *ct2,
+                             size_t ctlen) {
+    int ret = 1, ret_rev = 1, idx = 0;
+
+    if ((idx = oqs_get_idx(kemalg_name)) >= OSSL_NELEM(info_classical)) {
+        return 0;
+    }
+
+    ret_rev &= memcmp(sec1 + seclen - info_classical[idx].sec_len,
+                      sec2 + seclen - info_classical[idx].sec_len,
+                      info_classical[idx].sec_len) == 0 &&
+               memcmp(ct1 + ctlen - info_classical[idx].pubkey_len,
+                      ct2 + ctlen - info_classical[idx].pubkey_len,
+                      info_classical[idx].pubkey_len) == 0;
+
+    ret &= memcmp(sec1, sec2, info_classical[idx].sec_len) == 0 &&
+           memcmp(ct1, ct2, info_classical[idx].pubkey_len) == 0;
+
+    return ret || ret_rev;
+}
+
+/** \brief Compares the classical SIG elemets of two Sign executions
+ *
+ * \param sigalg_name algorithm name.
+ * \param sig1 A signature.
+ * \param sig1len Signature length.
+ * \param sig2 A signature.
+ * \param sig2len Signature length.
+ *
+ * \return 1 if the compare operation is successful, else 0. */
+static int oqs_cmp_sig_elems(const char *sigalg_name, const unsigned char *sig1,
+                             size_t sig1len, const unsigned char *sig2,
+                             size_t sig2len) {
+    int ret = 1, ret_rev = 1, idx;
+
+    if ((idx = oqs_get_idx(sigalg_name)) >= OSSL_NELEM(info_classical)) {
```

Comment:
nit: Why not include the sanity (length) check within the "..._get_idx" function (e.g., making it return -1 if any error occurred?)